source/tryGoal/component/tryGoalController.js

```diff
@@ -335,6 +265,14 @@ function($scope, agentMemory, $compile, $stateParams, config, GoalAPI, ConfigAPI
                 }
                 responseObject.push(temp);
             } else { //Second form
+				if (question.answer.selection.currentValue) {  //currentValue is used by some plural inputs where having it's value in the selection array is awkward.
+					question.answer.selection.push(question.answer.selection.currentValue);
+					delete question.answer.selection.currentValue;
+				}
+				if ('typeof', typeof (question.answer.selection) == 'string') {
```

Comment:
This doesn't look quite right to me. Was the first 'typeof' meant to be checking something?